florist/api/routes/server/training.py

```diff
@@ -43,12 +43,13 @@ async def start(job_id: str, request: Request, background_tasks: BackgroundTasks
         If not successful, returns the appropriate error code with a JSON with the format below:
             {"error": <error message>}
     """
+    job = None
     try:
         job = await Job.find_by_id(job_id, request.app.database)
 
         assert job is not None, f"Job with id {job_id} not found."
```

Comment:
If move the `try` clause below this check you can get rid of all the `if` conditions inside the excepts:
```python
job = await Job.find_by_id(job_id, request.app.database)

if job is None:
    return JSONResponse(content={"error": f"Job with id {job_id} not found."}, status_code=400)

try:

    ...

except AssertionError as err:
    await job.set_status(JobStatus.FINISHED_WITH_ERROR, request.app.database)
    return JSONResponse(content={"error": str(err)}, status_code=400)

except Exception as ex:
    LOGGER.exception(ex)
    await job.set_status(JobStatus.FINISHED_WITH_ERROR, request.app.database)
    return JSONResponse({"error": str(ex)}, status_code=500)
```
The only downside is having an uncaught exception at `Job.find_by_id` but I believe FastAPI has some basic handling in place for that.

Reply:
I had to wrap the find_by_id call in a try except to keep the tests passing. Let me know if this is okay.

Reply:
I see. If that's the case I think the old code is better, actually. Thanks for trying that out.